components/calendar/src/gregorian.rs

```diff
@@ -233,3 +233,215 @@ pub(crate) fn year_as_gregorian(year: i32) -> types::FormattableYear {
         }
     }
 }
+
+#[cfg(test)]
+mod gregorian_negagtive_nums_tests {
+
+    use icu::calendar::{gregorian::Gregorian, types::Era, Date, Iso};
+    use tinystr::tinystr;
+
+    #[derive(Debug)]
+    struct TestCase {
+        fixed_date: i32,
+        iso_year: i32,
+        iso_month: u8,
+        iso_day: u8,
+        expected_year: i32,
+        expected_era: Era,
+        expected_month: u32,
+        expected_day: u32,
+    }
+
+    fn check_test_case(case: TestCase) {
+        let iso_from_fixed: Date<Iso> = Iso::iso_from_fixed(case.fixed_date);
+        let greg_date_from_fixed: Date<Gregorian> = Date::new_from_iso(iso_from_fixed, Gregorian);
+        assert_eq!(greg_date_from_fixed.year().number, case.expected_year,
+            "Failed year check from fixed: {case:?}\nISO: {iso_from_fixed:?}\nGreg: {greg_date_from_fixed:?}");
+        assert_eq!(greg_date_from_fixed.year().era, case.expected_era,
+            "Failed era check from fixed: {case:?}\nISO: {iso_from_fixed:?}\nGreg: {greg_date_from_fixed:?}");
+        assert_eq!(greg_date_from_fixed.month().ordinal, case.expected_month,
+            "Failed month check from fixed: {case:?}\nISO: {iso_from_fixed:?}\nGreg: {greg_date_from_fixed:?}");
+        assert_eq!(greg_date_from_fixed.day_of_month().0, case.expected_day,
+            "Failed day check from fixed: {case:?}\nISO: {iso_from_fixed:?}\nGreg: {greg_date_from_fixed:?}");
+
+        let iso_date_man: Date<Iso> =
+            Date::try_new_iso_date(case.iso_year, case.iso_month, case.iso_day)
+                .expect("Failed to initialize ISO date for {case:?}");
+        let greg_date_man: Date<Gregorian> = Date::new_from_iso(iso_date_man, Gregorian);
```

Comment:
Suggestion: just check that the `_man` values are the same as the `_from_fixed` values

Reply:
done